### ISSUE TITLE
Replace the JSON diff report in the log with the new ASCII table version

### ISSUE THREAD

**Issue by tomakehurst** [tomakehurst/wiremock#799]

**Comment by HenningWaack** [comment]
Hi @tomakehurst.

Love Wiremock, thanks for maintaining it!

One issue I would like to address is the diff view for JSON body. It is quiet hard to read in this split view, the differences are difficult to spot, and it is cumbersome to copy the seperate JSON contents into an editor to do the compare there (because of the split view).

I was looking into alternatives views for this. One approach I find quite interesting is using JSON Patch API to compare two JSON docs, which produces an structured output of differences, which could then be transformed into a more readable form (like a git diff, or something similar). A nice blog article describing this approach is here: https://cassiomolin.com/programming/comparing-json-documents-in-java-with-jsonp/

What's your take? Are you still interested in enhancing the diff view? What would be a good approach from your point of view?

**Comment by tomakehurst** [comment]
I think there's definitely room for improvement in this area, although this particular issue is probably obsolete.

I guess you're suggesting we switch to an alternative way of presenting when two JSON bodies differ i.e. rather than just showing them both in full we show a diff summary?

Incidentally, we actually used to use JSONPatch internally when calculating the differences for matching (but ended up find the JSON assertions library was a better fit).

**Comment by HenningWaack** [comment]
Yes, an alternative way would be nice to display differences in JSON body. The other diff-views, e.g. for http status, headers etc. are fine I guess, but the diff in JSON is hard to read. 

I would suggest displaying the JSON bodies in two ways: received and expected JSON body below each other to support easy copy from console (into IDE for advanced compare features), and something like JSON Patch does or something along the lines of a GIT diff.

For something like the JSON Patch diff a standard lib from Apache or JBoss could be used (and would need to be bundled with wiremock). Something like a GIT diff could be custom code, I would hope that this would not be to complex. But this and possible other approaches should be evaluated somewhat more deeply.

**Comment by tomakehurst** [comment]
There are several types of use that need to be accommodated, which can be tricky given it's not always possible to know which one you're in:
1. Output from a test tool.
2. Output from a test tool into an IntelliJ terminal view (which parses specific forms of output and presents its own diff dialogue).
3.  Console output (might be streamed into a log) when running standalone, viewed directly.
4. 3. but (as you suggest) copy/pasted into an editor to take advantage of its diff viewer.

WireMock currently supports the first 3 cases, attempting to guess at the best diff rendering for the context, so one option would be to add yours as an additional strategy that could be opted into via a config flag.
